glue-codes/openfast-cpp/src/FAST_Prog.cpp

```diff
@@ -8,25 +8,44 @@ inline bool checkFileExists(const std::string& name) {
   return (stat (name.c_str(), &buffer) == 0); 
 }
 
+/// Optionally read in a value from a yaml node if present, else set it to a default value. Copied from github.com/Exawind/nalu-wind/include/NaluParsing.h
+template<typename T>
+void get_if_present(const YAML::Node & node, const std::string& key, T& result, const T& default_if_not_present = T())
+{
+    if (node[key]) {
+        const YAML::Node value = node[key];
+        result = value.as<T>();
+    }
+    else {
+        int rank;
+        int iErr = MPI_Comm_rank( MPI_COMM_WORLD, &rank);
+        if(!rank)
+            std::cout << key << " is missing in the input file. Proceeding with assumption " << key << " = " << default_if_not_present << std::endl ;
+        result = default_if_not_present;
+    }
+}
+
+/// Read a 'key' from a yaml node if it exists, else throw an error
+template<typename T>
+void get_required(const YAML::Node & node, const std::string& key, T& result)
+{
+    if (node[key]) {
+        const YAML::Node value = node[key];
+        result = value.as<T>();
+    }
+    else    {
+        throw std::runtime_error("Error: parsing missing required key: " + key);
+    }
+}
+
 void readTurbineData(int iTurb, fast::fastInputs & fi, YAML::Node turbNode) {
 
   //Read turbine data for a given turbine using the YAML node
-  if (turbNode["turb_id"]) {
-      fi.globTurbineData[iTurb].TurbID = turbNode["turb_id"].as<int>();
-  } else {
-      turbNode["turb_id"] = iTurb;
-  }
-  if (turbNode["FAST_input_filename"]) {
-      fi.globTurbineData[iTurb].FASTInputFileName = turbNode["FAST_input_filename"].as<std::string>() ;  
-  } else {
-      fi.globTurbineData[iTurb].FASTInputFileName = "";
-  }
-  if (turbNode["restart_filename"]) {
-      fi.globTurbineData[iTurb].FASTRestartFileName = turbNode["restart_filename"].as<std::string>() ;
-  } else {
-      fi.globTurbineData[iTurb].FASTRestartFileName = "";
-  }
-  if ( (fi.globTurbineData[iTurb].FASTRestartFileName == "") && (fi.globTurbineData[iTurb].FASTInputFileName == "") )
+  get_if_present(turbNode, "turb_id", fi.globTurbineData[iTurb].TurbID, iTurb);
+  std::string emptyString = "";  
```

Comment:
Since you have a default `T()`, you don't need to pass `emptyString` in each call.

Reply:
Ah.. yes. I suppose I have to create another `get_if_present` with just 3 arguments.

Reply:
You misunderstood me. The definition of `get_if_present` is

```
void get_if_present(const YAML::Node & node, const std::string& key, T& result, const T& default_if_not_present = T())
```

Which means if you call it with 3 arguments, it will default to empty string when `T = std::string`. You *don't* have to create another function with 3 arguments.

Reply:
Ah.. got it. Thanks. Fixed in  https://github.com/OpenFAST/openfast/pull/145/commits/4f16f1b50b147880f05a7fda054d82ff5ec06801